src/Symfony/Component/DependencyInjection/Compiler/AnalyzeServiceReferencesPass.php

```diff
@@ -101,6 +112,19 @@ protected function processValue($value, $isRoot = false)
                 ContainerInterface::IGNORE_ON_UNINITIALIZED_REFERENCE === $value->getInvalidBehavior()
             );
 
+            if ($this->inExpression) {
+                $this->inExpression = false;
```

Comment:
why making it `false` again ? Wouldn't this break in case an expression is referencing multiple services ?